### ISSUE TITLE
Training, Validation, Test  Dataset Split

### ISSUE THREAD

**Issue by marwankefah** [XMed-Lab/EPL_SemiDG#1]
This code is a great implementation of the research article that you published. I read through the code, and it is easy to read/understand, but I have three inquiries.

1) Are the unique images in the training set the same as in the validation set with no split? 
because I can see "val_dataset." is equal to the "label_dataset" before you duplicate the "label_dataset" multiple times to "new_label_dataset."

2) if the first question is yes, then are you saving the model every time it surpasses itself on the "val_dataset." which the model already sees during training?

3) The results published in the article are that after early stopping of the model and blindly evaluating it on the test set directly or is it the best dice score that the model output throughout the training process?

Thank you so much,

**Comment by nekomiao123** [comment]
Hi! I simply choose the best dice score as my result.

**Comment by marwankefah** [comment]
How can it not be evaluation bias on the test set?
and why the Val dataset has the same unique examples as the labeled dataset or is there an error in the split?

val_dataset = ConcatDataset(
        [domain_1_labeled_dataset, domain_2_labeled_dataset, domain_3_labeled_dataset])

label_dataset = ConcatDataset(
        [domain_1_labeled_dataset, domain_2_labeled_dataset, domain_3_labeled_dataset])

**Comment by nekomiao123** [comment]
I follow the [previous work](https://github.com/xxxliu95/DGNet) to conduct experiments. And I tried to split the val dataset from the training dataset, but the labeled ratio is too low(0.02 or 0.05), which means I only have several images in the val dataset.

**Comment by marwankefah** [comment]
Yes, I saw this now, and it is fair enough for comparison then. thank you a lot for letting me know
